### ISSUE TITLE
messages Endpoints

### ISSUE THREAD

**Issue by chris-rogala** [etsy/open-api#171]
**Is your feature request related to a problem? Please describe.**
Our shop receives 100-200 messages a weekend during our higher volume seasons. We easily spend 20--30 man hours a week processing messages. During our higher volume days, we easily receive well over 100 messages a day. A large portion of the messages can be grouped and a saved response is sent. For instance, we receive a lot of messages where people are checking on the ship-by-date or shipping address. We also have people submit photos for creating their custom artwork. 

In rare occasions, we have customers reach out to us about a negative experience or an issue with their package. The messages have different levels of severity and also have standard replies. 

**Describe the solution you'd like**
It would be extremely useful to have access to an endpoint that can get messages and reply to messages. While having access to send messages to customers with open order (e.g. a message reminding the customer to send a photo or a message to send the proof for an order) would be very useful, I know that Etsy is concerned about spam messages. That being said, having an endpoint only for unread messages and would only allow for replies to be sent for unread messages would be huge. This would greatly reduce the risk of developers creating solutions that spam the Esty customers. This solution would greatly reduce the time we spend managing messages and would provide customers with a better customer experience since they would receive a reply faster. Ideally, being able to message customers with open orders would even reduce our time spent on messages and provide a better customer experience (e.g. a friendly reminder to send their photo or respond to a proof.)

**Describe alternatives you've considered**
Since our product requires customers to submit photos for review, I have developed a secure customer portal that provides the customer a friendly wizard to submit their photo and customization requests. Once the pandemic hit, I was forced to delay that project and do some consulting with a previous client.

**Additional context**
Amazon provides a way to send automated messages. Within Amazon, we will sell 50+ custom items during the holiday season. I am able to send all 50 proofs in a matter of seconds and this ensure our lead time remains at 1-2 business days even during the holiday rush. On Etsy, I am not able to do this, so our lead times can get extended by several days and Etsy can lose sales to other platforms.

**Comment by annedirkse** [comment]
+1 for this. The API is the only way that we can manage the volume of orders that we get from our shop, but messages are a critical pain point for us. We use the API to gather and categorize orders to proof and currently send by e-mail because that is the only way we can collect approvals/revisions in a semi-automated way. I would add to the above that it would be useful to be able to send messages via the API to customers with open orders, along with the ability to reply to incoming messages.

The messenger interface is problematic especially given Etsy's new "Star Seller" program which requires a response to every Etsy message within 24 hours. The time required to respond to Etsy messages is significant, and we cannot hire someone to help process those messages without giving out our master account login and password which is obviously extremely problematic.

**Comment by chris-rogala** [comment]
@annedirkse this would also help to solve how buggy the messaging system is in Etsy. Around the holidays, we live off of tagging messages. We have run into numerous issues with tags not persisting. Additionally, we notice during higher volume season that message are not delivered. We noticed proofs did not go through, even though they were sent and even have customers tell us frequently that they replied to a message and we never received it. Having the ability to send a message through the API and then validate the message was sent would help to ensure packages are shipped on time. For custom orders that require proof approval, this is a must and core to our business.

**Comment by annedirkse** [comment]
Yes, exactly. When we did our proofing exclusively through Etsy messenger we'd have many, many customers replying to the email from messenger which goes to a no-reply address _unless_ the user is a guest. That's its own separate problem, but back then I just wrote scripts to scan our email for messages from users we had outstanding proofs for, add the content of the message to the proof in our system, and we had a reply function that opened the url to reply to that user in an iFrame. It was so ridiculously kludgey but it saved us hours each day.

We do follow up with messenger when we don't get a response by email. But at least we know when an e-mail bounces. It would be very nice to know that a message was received and the timestamp being exposed by the api in addition to the message would allow us to do our own internal metrics, i.e. highlight messages over x minutes old during standard business hours.

Etsy's product teams want us to use the messenger tool, it makes sense for the brand, so it should be as easy from the API as it is to e-mail.

**Comment by XAMelleOH** [comment]
Any feedback from Etsy team on this topic? 

FYI @etsyachristensen @ataetsy @pma-etsy @rpinedaetsy @etsy @Vaibhav @manjuraj

**Comment by sarahschroeter** [comment]
+1 for this feature, our shop receives roughly 200+ messages a day, and it is not sustainable to manage that volume in the Etsy messenger without more tools. We have 4 cs agents in this queue daily, which we are not easily able to monitor each agent's volume, automatically tag issues using key words, etc. Our shop also sells on other marketplaces which all integrate with Zendesk, leaving Etsy as a sort of data black hole. We would love to be able to pull customer insights from our messages to better improve our products, get sentiment, and gather ticket counts to make smart decisions for our business needs as we grow.

**Comment by flmu** [comment]
@etsyachristensen Any updates on this topic?

**Comment by turboturtleD** [comment]
Andrew is no longer working for Etsy.  I think you want @nateswart.  However, I don’t see us getting anything from Etsy in the realm of messaging.  Big hopes for v3 kinda fell flat on this topic specifically.  Read only would have been at least a great start.
